sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosMultiHashTest.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ *
+ */
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.HttpConstants;
+import com.azure.cosmos.implementation.InternalObjectNode;
+import com.azure.cosmos.implementation.apachecommons.lang.StringUtils;
+import com.azure.cosmos.implementation.routing.PartitionKeyInternal;
+import com.azure.cosmos.models.*;
+import com.azure.cosmos.rx.TestSuiteBase;
+import com.azure.cosmos.util.CosmosPagedIterable;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class CosmosMultiHashTest extends TestSuiteBase {
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private String preExistingDatabaseId = CosmosDatabaseForTest.generateId();
+    private CosmosClient client;
+    private CosmosDatabase createdDatabase;
+    private CosmosContainer createdMultiHashContainer;
+
+    @Factory(dataProvider = "clientBuilders")
+    public CosmosMultiHashTest(CosmosClientBuilder clientBuilder) {
```

Comment:
Could you also add some tests for bulk and batch with subpartitioning?

Reply:
Will cover these in the next iteration.